core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -3054,17 +3069,26 @@ private void addPropertiesWithID(Connection dbConnection, String userID, Map<Str
 
             for (Map.Entry<String, String> entry : userAttributes.entrySet()) {
                 String propertyName = entry.getKey();
-                String propertyValue = entry.getValue();
-                if (sqlStmt.contains(UserCoreConstants.UM_TENANT_COLUMN)) {
-                    if (UserCoreConstants.OPENEDGE_TYPE.equals(type)) {
-                        batchUpdateStringValuesToDatabase(prepStmt, propertyName, propertyValue, profileName, tenantId,
-                                userID, tenantId);
+                List<String> propertyValues = new ArrayList<>();
+                if (multiValuedAttributes.contains(propertyName)) {
+                    String[] values = entry.getValue().split(multiAttributeSeparator);
+                    propertyValues.addAll(Arrays.asList(values));
+                } else {
+                    propertyValues.add(entry.getValue());
+                }
+
+                for (String propertyValue : propertyValues) {
```

Comment:
Check how multi valued claims should be inserted in identity-db

Reply:
The identity claim handling is handled from different PR and there are some points to be concluded about improving the unique key constraint to store entries with same key as we have done for the user attributes tables.